### PR TITLE
Mobile App: fix bug stuck button send message

### PR DESCRIPTION
**Pull request by huylexuan44**

### CODE REVIEW

File: apps/mobile/src/app/screens/home/homedrawer/ChannelMessages.tsx [hunk 1]
```diff
@@ -116,41 +112,48 @@ const ChannelMessages = React.memo(({ channelId, clanId, channelLabel, mode }: C
...
 			const offsetY = event.nativeEvent.contentOffset.y;
 			const threshold = 300; // Adjust this value to determine when to show the button
-
-			if (offsetY > threshold && !showScrollToBottomButton) {
-				setShowScrollToBottomButton(true);
-			} else if (offsetY <= threshold && showScrollToBottomButton) {
+			if (offsetY > threshold && showScrollToBottomButton) {
 				setShowScrollToBottomButton(false);
+			} else if (offsetY <= threshold && !showScrollToBottomButton) {
+				setShowScrollToBottomButton(true);
```
Comment (phuocnguyenncc) [inline apps/mobile/src/app/screens/home/homedrawer/ChannelMessages.tsx:145]: chắc là mình remove cái state này dùng cái `isViewingOldMessage` để show icon jump to end anh

File: apps/mobile/src/app/screens/home/homedrawer/ChannelMessages.tsx [hunk 2]
```diff
@@ -116,41 +112,48 @@ const ChannelMessages = React.memo(({ channelId, clanId, channelLabel, mode }: C
...
-		flatListRef.current.scrollToOffset({ animated: true, offset: 0 });
-	};
-
-	const ViewLoadMore = () => {
-		return (
-			<View style={styles.loadMoreChannelMessage}>
-				<ActivityIndicator size="large" color={Colors.tertiary} />
-			</View>
-		);
+		flatListRef.current.scrollToEnd({ animated: true });
```
Comment (phuocnguyenncc) [inline apps/mobile/src/app/screens/home/homedrawer/ChannelMessages.tsx:156]: update action in function scroll bottom like component `ChannelJumpToPresent` in web

File: apps/mobile/src/app/screens/home/homedrawer/ChannelMessages.tsx [hunk 3]
```diff
@@ -161,19 +164,14 @@ const ChannelMessages = React.memo(({ channelId, clanId, channelLabel, mode }: C
...
 	const jumpToRepliedMessage = useCallback(
 		(messageId: string) => {
-			const indexToJump = dataReverse.findIndex((message) => message === messageId);
+			const indexToJump = messages.findIndex((message) => message.id === messageId);
 			if (indexToJump !== -1 && flatListRef.current) {
 				flatListRef.current.scrollToIndex({ animated: true, index: indexToJump - 1 });
 			}
 		},
-		[dataReverse],
+		[messages],
```
Comment (phuocnguyenncc) [inline apps/mobile/src/app/screens/home/homedrawer/ChannelMessages.tsx:174]: messages.id

File: apps/mobile/src/app/screens/home/homedrawer/MessageItem.tsx [hunk 4]
```diff
@@ -399,6 +395,10 @@ const MessageItem = React.memo((props: MessageItemProps) => {
 			<NewMessageRedLine channelId={props?.channelId} messageId={props?.messageId} isEdited={isEdited} />
 		</Animated.View>
 	);
-});
+},
+  (prevProps, nextProps) => {
+	return prevProps.messageId === nextProps.messageId;
+  }
+);
```
Comment (phuocnguyenncc) [inline apps/mobile/src/app/screens/home/homedrawer/MessageItem.tsx]: compare re-render  `message.id` and `mesage.update_time` to cover case message edited

File: apps/mobile/src/app/screens/home/homedrawer/ChannelMessages.tsx (hunk 3 above)
Comment (huylexuan44) [reply apps/mobile/src/app/screens/home/homedrawer/ChannelMessages.tsx:174]: function inner channelMessages not MessageItem

**Pull request merged**